manifests/server/root_password.pp

```diff
@@ -17,6 +17,10 @@
       owner   => 'root',
       mode    => '0600',
     }
+    # show_diff was added with puppet 3.0
+    if versioncmp($::puppetversion, '3.0') <= 0 {
+      File["${::root_home}/.my.cnf"] { show_diff => false }
```

Comment:
this is not a default. this is actually setting the value and will cause a conflict if done twice.